hnbex/__init__.py

```diff
@@ -0,0 +1,74 @@
+import json
+import re
+import requests
+from datetime import datetime
+from decimal import Decimal
+
+
+class Rate(object):
+
+    RE_VALIDATION_DICT = {
+        'currency_code': r'^[A-Z]{3}$',
+        'buying_rate': r'^[0-9]+.[0-9]{6}$',
+        'selling_rate': r'^[0-9]+.[0-9]{6}$',
+        'median_rate': r'^[0-9]+.[0-9]{6}$',
+    }
+
+    URL = 'http://hnbex.eu/api/v1/rates/daily/'
+
+    def __init__(self, currency_code, rate_date, buying_rate, selling_rate,
+                 median_rate, unit_value):
+
+        self._validate(currency_code, rate_date, buying_rate, selling_rate,
+                       median_rate, unit_value)
+
+        self.code = currency_code
+        self.date = rate_date
+        self.buy = Decimal(buying_rate)
+        self.sell = Decimal(selling_rate)
+        self.median = Decimal(median_rate)
+        self.unit_value = unit_value
+
+    def _validate(self, currency_code, rate_date, buying_rate, selling_rate,
+                  median_rate, unit_value):
+
+        for var, regex in self.RE_VALIDATION_DICT.items():
+            if re.match(regex, eval(var)) is None:
+                raise self.ValidationException('{}: wrong format'.format(var))
+
+        if unit_value not in [1, 100]:
+            raise self.ValidationException('unit_value must be 1 or 100')
```

Comment:
It’s fine to use `ValueError` here, since the meaning is the same. In Python we’re encouraged to reuse the built-in exceptions if their meaning matches the desired one. Here, the meaning is “the argument value is not correct/applicable”, which is the exact meaning of ValueError.